pkgs/stdenv/adapters.nix

```diff
@@ -60,12 +60,13 @@ rec {
       mkDerivationFromStdenv = withOldMkDerivation old (stdenv: mkDerivationSuper: args:
       if stdenv.hostPlatform.isDarwin
       then throw "Cannot build fully static binaries on Darwin/macOS"
-      else (mkDerivationSuper args).overrideAttrs(finalAttrs: {
-        NIX_CFLAGS_LINK = toString (finalAttrs.NIX_CFLAGS_LINK or "") + " -static";
-      } // lib.optionalAttrs (!(finalAttrs.dontAddStaticConfigureFlags or false)) {
-        configureFlags = (finalAttrs.configureFlags or []) ++ [
-            "--disable-shared" # brrr...
-          ];
+      else (mkDerivationSuper args).overrideAttrs (args: {
+        NIX_CFLAGS_LINK = toString (args.NIX_CFLAGS_LINK or "") + " -static";
+      } // lib.optionalAttrs (!(args.dontAddStaticConfigureFlags or false)) {
+        configureFlags = (args.configureFlags or []) ++ [
+          "--disable-shared" # brrr...
+        ];
+        cmakeFlags = (args.cmakeFlags or []) ++ ["-DCMAKE_SKIP_INSTALL_RPATH=On"];
```

Comment:
It's unneeded to have this under the `dontAddStaticConfigureFlags` because this is a cmake builtin flag

`dontAddStaticConfigureFlags` is there for configure systems that error when there's a flag they don't recognise

Reply:
My thinking is that it can still be useful to have an easy way to toggle it off. Perhaps this could cause other downstream errors and a user might want to be able to turn it off in the course of investigating. Happy to change it though if you insist. Let me know.

Reply:
`makeStaticLibraries` has it under `dontAddStaticConfigureFlags` too so keep it as is